editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx

```diff
@@ -954,6 +954,45 @@ export function useClearKeyboardInteraction(editorStoreRef: {
   }, [dispatch, editorStoreRef])
 }
 
+export function useClearStaticReparentInteraction(editorStoreRef: {
+  readonly current: EditorStorePatched
+}): () => void {
+  const dispatch = useDispatch()
+  return React.useCallback(() => {
+    const clearStaticReparentInteractionOnMouseDown = () => {
+      window.removeEventListener('mousedown', clearStaticReparentInteractionOnMouseDown)
+      if (
+        editorStoreRef.current.editor.canvas.interactionSession?.interactionData.type ===
+        'STATIC_REPARENT'
+      ) {
+        dispatch([CanvasActions.clearInteractionSession(true)], 'everyone')
+      }
+    }
+
+    window.addEventListener('mousedown', clearStaticReparentInteractionOnMouseDown, {
```

Comment:
I think it's possible for these event listeners to leak and then potentially duplicate, we often use `useEffect` for these as we can return a cleanup function that calls `removeEventListener`.

Reply:
added a useEffect to clean the handlers up in https://github.com/concrete-utopia/utopia/pull/3786/commits/667daff4af8da84ad113f9f721d8a9035d31420e